src/create_venv.sh

```diff
@@ -1,136 +1,206 @@
 #!/usr/bin/env bash
 
+# Copyright (c) YugaByte, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations
+# under the License.
+#
+
+[[ "${_YB_CREATE_VENV_SHLIB:=""}" == "yes" ]] && return 0
+_YB_CREATE_VENV_SHLIB=yes
+VERBOSE=${VERBOSE:-false}
 
-# Use SHA of requirements.txt as a comment in requirements_frozen.txt
-# 
+set -e -u -o pipefail
+
+DIR="${BASH_SOURCE%/*}"
+if [[ ! -d "$DIR" ]]; then DIR="$PWD"; fi
+# shellcheck disable=SC1091,SC1090
+. "${DIR}"/logger.sh
+# shellcheck disable=SC1091,SC1090
+. "${DIR}"/os.sh
+# shellcheck disable=SC1091,SC1090
+. "${DIR}"/detect_python.sh
+
+# -------------------------------------------------------------------------------------------------
+# Global variables used in this module
+# -------------------------------------------------------------------------------------------------
+YB_BUILD_STRICT=${YB_BUILD_STRICT:-false}
+YB_RECREATE_VIRTUALENV=${YB_RECREATE_VIRTUALENV:-false}
+YB_VENV_BASE_DIR=${YB_VENV_BASE_DIR:-~/.venv/yb}
+YB_USE_TOP_LEVEL_VENV=${YB_USE_TOP_LEVEL_VENV:-false}
+
+verbose "Using YB_PYTHON_VERSION=${YB_PYTHON_VERSION}"
+# shellcheck disable=SC2154
+verbose "Using ${yb_python_interpreter} (${yb_python_version_actual})"
+
+# -------------------------------------------------------------------------------------------------
+# Internal functions used in this module.  These shouldn't be called directly outside this module.
+# -------------------------------------------------------------------------------------------------
 
 function text_file_sha() {
   local file="${1}"
   local tmp
   tmp="$(sort -u <<<"$(grep -v '^#' "${file}")")"
-  awk '{print $1}'<<<"$(sha256sum <<<"${tmp}")"
+  # shellcheck disable=SC2154
+  awk '{print $1}'<<<"$(${yb_sha256sum} <<<"${tmp}")"
 }
 
-set -e -u -o pipefail
-
-YB_PYTHON_VERSION=${YB_PYTHON_VERSION:-3.7}
-py_major_version=$(cut -f1 -d. <<<"${YB_PYTHON_VERSION}")
-
-YB_VENV_BASE_DIR=${YB_VENV_BASE_DIR:-~/.venv/yb}
-YB_REFRESH_VENV=${YB_REFRESH_VENV:-false}
-
-echo "Using YB_PYTHON_VERSION=${YB_PYTHON_VERSION}"
-
-set +u
-if [[ -n "${VIRTUAL_ENV}" ]] && [[ -d "${VIRTUAL_ENV}" ]]; then
-  echo "Please exit your current venv first."
-  echo "type 'deactivate' in your shell"
-  exit 1
-fi
-set -u
-
-py_cmd=''
-for cmd in "python${YB_PYTHON_VERSION}" "python${py_major_version}" "python"; do
-  if cmd="$(command -v "${cmd}")"; then
-    if "${cmd}" --version 2>&1 | grep "${YB_PYTHON_VERSION}"; then
-      py_cmd="${cmd}"
-      break
+function needs_refreeze() {
+  local reqs_sha="${1}"
```

Comment:
I think `local reqs_sha=$1` would be sufficient (the extra double quotes and curly braces are redundant, at least in Bash). Let me know if this is more compatible with other shells.